core/shared/src/main/scala/scalaz/zio/internal/OneShot.scala

```diff
@@ -31,11 +31,13 @@ private[zio] class OneShot[A] private (@volatile var value: A) {
   /**
    * Retrieves the value of the variable, blocking if necessary.
    */
-  final def get: A = {
-    while (value == null) {
+  final def get(timeout: Long = Long.MaxValue): A = {
+    if (value == null) {
       this.synchronized {
-        if (value == null) this.wait()
+        if (value == null) this.wait(timeout)
```

Comment:
The Javadoc for `wait` indicates that spurious wake ups are possible (https://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#wait(long)).

I see the reason for adding this is so `unsafeRunSync` can be called with a timeout (nowhere else if I'm not mistaken?). You previously mentioned that you would like for the RTS to be like a "microkernel" for executing `IO`s. Since timeout can be expressed in terms of `IO`, is this following in that vein?

Reply:
Debug code. Fixed, thanks!

Reply:
`wait` can be woken up without a corresponding `notify`/`notifyAll` according to the javadocs, in which case this method would throw before the actual timeout expired. is that ok ?

Reply:
I need to fix that too. I changed it from the old while loop. Followup PR.